### DISCUSSION TITLE
Alpha version (test)

### DISCUSSION THREAD

**Discussion by remittor** [remittor/zapret-openwrt#1] [Ideas]
Вот так выглядит альфа версия модуля `luci-app-zapret`:

![image](https://github.com/user-attachments/assets/f3d0182a-71a8-478c-bb35-4d02ff650fda)

![image](https://github.com/user-attachments/assets/11fa9574-bae3-45b9-9576-8a9e5754b4d4)

![image](https://github.com/user-attachments/assets/933dc763-ba99-4ae8-8823-50939e43575e)

**Comment by NewUse** [comment]
а что за sync_config.sh, очень похоже на помесь велосипеда и костыля....

Зачем тебе нужен рефреш? В принципе, можно взять пример из https://github.com/muink/luci-app-dnsproxy/blob/master/htdocs/luci-static/resources/view/dnsproxy.js

**Reply by remittor** [reply]
> а что за sync_config.sh, очень похоже на помесь велосипеда и костыля....

Так сделано специально, что бы полностью отделить GUI часть от основной. Т.е. человек в любой момент может отказаться от GUI и просто править `/opt/zapret/config` вручную.

**Comment by samara1532** [comment]
а что там уникального что собрано только под определенную архитектуру?

**Reply by remittor** [reply]
> а что там уникального что собрано только под определенную архитектуру?

Я просто собрал бинари для тестов.
Следующий шаг - сделать автосборщик.

**Comment by remittor** [comment]
В релизе от 2024-10-16 пофиксил много ошибок касательно обновления ipk и удаления ipk.

При установке новой версии все бинари заменяются, а основной UCI-конфиг "/etc/config/zapret" мержится (старые параметры остаются, а новые добавляются).
При удалении пакета основной конфиг "/opt/zapret/config" бэкапится. При этом директория "/opt/zapret" не удаляется (это конечно спорное решение).

При бэкапе конфигурации всей OpenWrt в tgz файл складываются и все конфиги zapret (в том числе txt списки).

Новый скрин:
![image](https://github.com/user-attachments/assets/b4fd15ea-aead-4388-bf08-4d656c6727b3)

**Comment by ewgen198409** [comment]
Не плохо. Так более удобно. Еще прикрутить нормально статус сервиса и было бы нелохо видеть лог. А в идеале прикрутить blockcheck. Тогда цены пакету не будет. 
![image](https://github.com/user-attachments/assets/d3314cc4-3140-4457-b82c-fd7112a28b25)

**Reply by remittor** [reply]
> Еще прикрутить нормально статус сервиса и было бы нелохо видеть лог

сделано

**Comment by Fr0stmatic** [comment]
Неплохо для начала.
Поддерживаю пожелания выше.
И от себя — прошу прикрутить в luci-app переназначение портов. Для восстановления тех же звонков в Discord было бы полезно.
Ещё бы для сторонних листов придумать что-то более гибкое, нежели тупо 4 пустых cust*.txt (банально самому создавать нужное количество листов со своим названием).
Да и в целом причесать бы luci-app, ибо на прочих темах (особенно на стандатной Bootstrap) интерфейс крайне невыверенный

**Reply by remittor** [reply]
> прошу прикрутить в luci-app переназначение портов

В версии от 23 окт добавил такое на форму.
Вот только чую, что для нормальной работы нужно вручную раскоментировать нужные опции в файле /opt/zapret/config
Зря я в этом файле коментарии оставил....

**Comment by remittor** [comment]
Прикручиваю autohostlist:

![image](https://github.com/user-attachments/assets/1161e13e-f4ed-4c64-a863-ca81c7d7c7fd)

**Comment by remittor** [comment]
Завершил адаптацию под новую v66 версию:

![image](https://github.com/user-attachments/assets/86e3b3fd-b4d0-42ce-ad8b-27471abffb10)

**Comment by barmaley6699** [comment]
не всегда запускается после перезагрузки.

**Reply by remittor** [reply]
Воспроизвести не смог. В этой части вопрос не ко мне, а к автору оригинала, т.к. скрипт автозапуска мною никак не правился. Правится только основной конфиг.

**Reply by barmaley6699** [reply]
даже не после перезагрузки, а сам по себе. От автора такого не замечал ни разу. С августа пользуюсь. В последней версии появилась такая беда.

**Reply by remittor** [reply]
Ну вот читайте ветку: https://github.com/bol-van/zapret/issues/631
Проверьте права на файлы.

**Reply by barmaley6699** [reply]
0 */12 * * * /opt/zapret/init.d/openwrt/zapret restart 
написал в планировщик. Скажите я правильно написал? Только осваиваю openwrt. И у автора в планировщике тоже написано. Точно не помню как, надо установить посмотреть.

**Reply by barmaley6699** [reply]
права правильные

**Comment by tsa174** [comment]
Хотелось бы иметь возможность запускать скрипты custom. Идея не плохая, но вот как с custom быть.

**Reply by remittor** [reply]
На то они и custom **скрипты**. Именно скрипты.
Вы предлагаете в WebUI добавить возможность создания кастомных скриптов в папке /opt/zapret ?
Как то бредово.

**Reply by tsa174** [reply]
А почему нет? Речь не про создание, а про запуск. Скрипты то есть уже! А вот как запускать у вас их, из за этого пришлось снести вашу сборку. С помощью них и только с помощью них по заявлению bolvan нужно решать дополнительные задачи. К примеру DISCORD, далее ZEROTIER. Работает и Discord voice и Zerotier сеть на мобильных операторах с помощью кастомных скриптов, а так получается без них кастрированный zapret!  Возможно у вас есть другое решение.

**Reply by remittor** [reply]
Да вы тот же самый конфиг, который прописывают эти кастомы, можете сейчас через WebUI прописать.

**Reply by tsa174** [reply]
Пытался прописать, но не заработало как нужно. Возможно что то не так делал. Да и нельзя так делать как пишет bol-van. Есть у него ветка на тему Discord там все разжевано.

**Comment by zxcursedead6** [comment]
Товарищи, приветствую. Очень не хватает кнопки сброса конфига для NFQWS-OPT
И как готовый конфиг переписать, чтобы дурачил только host-list. Тот же рутрекер работает, а хабр ложится ( err_ssl_protocol_error  )

**Reply by remittor** [reply]
> Очень не хватает кнопки сброса конфига для NFQWS-OPT

А кнопки Reset не хватает?

**Comment by sgholley** [comment]
Привет. Во первых спасибо за созданную web версию. Еще и получилось установить знакомому на роутер с отсутствующим USB. Но возник вопрос в ходе настройки, у меня не получилось как на версии entware установить фикс для дискорда, все пробовал как у себя, но результата ноль. Возможно это просто еще не поддерживается?
делал вот так:
1) Останавливаем zapret командой:
/opt/zapret/init.d/sysv/zapret stop

2) Копируем скрипт 50-discord командой:
cp -a /opt/zapret/init.d/sysv/custom.d.examples/50-discord /opt/zapret/init.d/sysv/custom.d/50-discord

3) Запускаем zapretкомандой:
/opt/zapret/init.d/sysv/zapret start

После этого будет поднят отдельный процесс NFQWS для модификации трафика на портах UDP 50000-50099. Стратегия модификации задается в переменной NFQWS_OPT_DESYNC_DISCORD этого скрипта.

**Reply by remittor** [reply]
Этот custom создан для лентяев и для тех, кто на основе него может создать свой кастом-скрипт.
У меня тут проект является не консольным, а WebUI ориентированным.
Поэтому правила для дискорда прописывайте через WebUI. Это возможно.

**Reply by sgholley** [reply]
> Этот custom создан для лентяев и для тех, кто на основе него может создать свой кастом-скрипт. У меня тут проект является не консольным, а WebUI ориентированным. Поэтому правила для дискорда прописывайте через WebUI. Это возможно.

Ну ладно уж для лентяев, удобство не есть лентяйство. Но все равно спасибо за ответ, буду сидеть разбираться дальше, потому что вручную возможно что-то не так делаю

**Comment by 2Grey** [comment]
Команда определения архитектуры через SSH:
```sh
grep ARCH /etc/openwrt_release | awk -F= '{print $2}'
```

**Comment by HayoiGlich** [comment]
Здравствуйте, помогите настроить данный запрет чтобы работал и дискорд и ютаб. Я пользуюсь для винды батником и там вот такие параметры. 
`@echo off
chcp 65001 >nul
:: 65001 - UTF-8

cd /d "%~dp0..\"
set BIN=%~dp0..\bin\

set LIST_TITLE=ZAPRET: Ultimate Fix ALT v2
set LIST_PATH=%~dp0..\lists\list-ultimate.txt
set DISCORD_IPSET_PATH=%~dp0..\lists\ipset-discord.txt

start "%LIST_TITLE%" /min "%BIN%winws.exe" ^
--wf-tcp=80,443 --wf-udp=443,50000-65535 ^
--filter-udp=443 --hostlist="%LIST_PATH%" --dpi-desync=fake --dpi-desync-repeats=6 --dpi-desync-fake-quic="%BIN%quic_initial_www_google_com.bin" --new ^
--filter-udp=50000-65535 --ipset="%DISCORD_IPSET_PATH%" --dpi-desync=fake --dpi-desync-any-protocol --dpi-desync-cutoff=d3 --dpi-desync-repeats=6 --new ^
--filter-tcp=80 --hostlist="%LIST_PATH%" --dpi-desync=fake,split2 --dpi-desync-autottl=2 --dpi-desync-fooling=md5sig --new ^
--filter-tcp=443 --hostlist="%LIST_PATH%" --dpi-desync=split2 --dpi-desync-split-seqovl=652 --dpi-desync-split-pos=2 --dpi-desync-split-seqovl-pattern="%BIN%tls_clienthello_www_google_com.bin"`

**Reply by ewgen198409** [reply]
Он работает искаропки

**Reply by HayoiGlich** [reply]
Ютуб не грузит, а дискорд не входит в голосовые

**Reply by ewgen198409** [reply]
Ставьте ещё dns crypt. Кажется так называется. Инфа есть на гите по установке zapret na openwrt

**Reply by ewgen198409** [reply]
Мой провайдер тоже по днсу режет.

**Reply by HayoiGlich** [reply]
https-dns-proxy я установил и что теперь делать, написать порты или оставить как есть?
